Add semantic-release

jeffreylo opened the pull request:
This PR adds semantic-release to ensure that the releases and tags are kept in sync with the project’s changes.

jeffreylo committed:
build: tidy dependencies

jeffreylo committed:
ci: add configuration for semantic-release

This patch adds semantic-release to the CI pipeline to ensure that
releases and tags are kept in sync with the project's changes.

johngibb reviewed (approved):
Cool! Does it actually create the tags for you?

jeffreylo committed:
docs: add CONTRIBUTING.md

This patch adds the contributing guidelines to ensure that
contributors author commit messages that reflect our project's
standards.

jeffreylo commented:
>Cool! Does it actually create the tags for you?

It does via deriving the tag version from the commit history.

It does mean that contributors should take care in introducing breaking changes as [Go’s semantic import versioning](https://github.com/golang/go/wiki/Modules#faqs--semantic-import-versioning) will need to be considered.
